tinygrad/ops.py

```diff
@@ -22,28 +23,22 @@ class LoadOps(Enum): EMPTY = auto(); RAND = auto(); CONST = auto(); FROM = auto(
 Op = Union[UnaryOps, BinaryOps, ReduceOps, MovementOps, LoadOps, TernaryOps]
 OpType = Union[Type[UnaryOps], Type[BinaryOps], Type[ReduceOps], Type[MovementOps], Type[LoadOps], Type[TernaryOps]]
 
+@dataclass(slots=True, weakref_slot=True)
 class LazyOp:
-  __slots__ = "op", "src", "arg", "buffers", "__weakref__"
   op: Op
   src: Tuple[Union[LazyOp, LazyBuffer], ...]
-  arg: Any
-  buffers: Tuple[LazyBuffer, ...]
+  arg: Any = None
+  buffers: Tuple[LazyBuffer, ...] = field(init=False)
 
-  def __init__(self, op: Op, src: Tuple[Union[LazyOp, LazyBuffer], ...], arg: Any = None):
-    self.op = op
-    self.src = src
-    self.arg = arg
+  def __post_init__(self):
```

Comment:
The behavior of this looks different

Reply:
I do not follow you.

Reply:
I mean do we still need the try?

The behavior looks different because I think it used to be all or nothing but now it's not.

Reply:
Seems to me that it still is?

Reply:
Outdated.